server/src/main/java/com/cloud/user/AccountManagerImpl.java

```diff
@@ -1847,6 +1832,29 @@ public boolean deleteUserAccount(long accountId) {
         return deleteAccount(account, callerUserId, caller);
     }
 
+    private boolean checkDeleteNeeded(AccountVO account, Account caller) {
```

Comment:
the method name is a bit confused.

if it returns `true`, then cleanup is not needed, right ?

Reply:
right, I will reverse the logic. This was refactorred a bit to criticlessly ;)

Reply:
@DaanHoogland 
maybe just rename the method

Reply:
did that (including reversing the logic as that makes for more intuitive naming)

Reply:
good, looks good to me